### ISSUE TITLE
[8.7.1] Randomized Truncation에 대한 이해

### ISSUE THREAD

**Issue by 1gyeol-KIM** [woowacampers/d2l-pytorch#7]
<img width="1100" alt="스크린샷 2021-02-19 오후 5 32 00" src="https://user-images.githubusercontent.com/42639690/108478972-7ec0d080-72d8-11eb-9f34-76d30fe6ee78.png">
식의 제타의 의미가 와닿지 않아서, random하게 truncation하는 과정도 이해가 잘 안되는 것 같습니다.

**Comment by staycozyboy** [comment]
제가 어떻게 이해했는지 적어볼께용

일단 regular truncation이 _focuses primarily on short-term influence_하기 때문에, 이를 완화시켜주고자(즉, long-term dependency를 잘 반영하기 위해) subsequence length를 random하게 정하는 방법이 고안된 것 같습니다. -> The length of each subsequence is sampled according to a specific probability distribution.

그 방법은 gradient computation flow에서 t step에서 t' step으로 넘어갈 때 특정한 확률에 따라 넘어갈 지 stop할 지를 결정하는 것입니다. 이때 balance를 맞추기 위해(먼 스텝까지 가게되더라도 gradient exploding or vanishing이 일어나지 않도록) 그 확률의 역수를 gradient flow에 곱해줍니다. 결과적으로 다음과 같이 back propagation이 진행됩니다.

![image](https://user-images.githubusercontent.com/37537248/109928702-11199900-7d09-11eb-8632-2a272fe2afe3.png)

자세한 사항은 [_Unbiasing Truncated Backpropagation Through Time_, 2017](https://arxiv.org/pdf/1705.08209.pdf)에서 확인할 수 있고, 추가적으로 좋은 probability distribution을 고르는 기준에 대한 설명도 있으니 참고하시면 될 것 같습니다. 개인적으로 d2l에서 좀 더 직관적으로 확인할 수 있는 것 같습니다. 다음과 같습니다 ㅎㅎ

![image](https://user-images.githubusercontent.com/37537248/109928851-38706600-7d09-11eb-82c1-1dc3fb3393ce.png)

**Comment by jiwon-ryu** [comment]
![image](https://user-images.githubusercontent.com/69613571/109930084-92bdf680-7d0a-11eb-8f0f-e247fca6da29.png)

계산의 편의를 위해 모든 dh/dW를 계산하지 않고 기댓값이 1인 제타를 도입해서 backprop을 한 것 같습니다
